cpg_workflows/stages/aip.py

```diff
@@ -0,0 +1,294 @@
+            f'python3 reanalysis/hail_filter_and_label.py '
+            f'--mt {input_mt} '
+            f'--panelapp {panelapp_json} '
+            f'--pedigree {local_ped} '
+            f'--vcf_out {str(expected_out["labelled_vcf"])}'
+        )
+
+        return self.make_outputs(dataset, data=expected_out, jobs=job)
+
+
+def _aip_summary_meta(
+    output_path: str,  # pylint: disable=W0613:unused-argument
+) -> dict[str, str]:
+    """
+    Add meta.type to custom analysis object
+    """
+    return {'type': 'aip_output_json'}
+
+
+@stage(
+    required_stages=RunHailFiltering,
+    analysis_type='custom',
+    analysis_keys=['summary_json'],
+    update_analysis_meta=_aip_summary_meta,
+)
+class ValidateMOI(DatasetStage):
+    """
+    run the labelled VCF -> results JSON stage
+    """
+
+    def expected_outputs(self, dataset: Dataset) -> dict[str, Path]:
+        return {'summary_json': dataset.prefix() / DATED_FOLDER / 'summary_output.json'}
+
+    def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput:
+        job = get_batch().new_job(f'AIP summary for {dataset.name}')
+        STANDARD.set_resources(job, ncpu=1, mem_gb=4.0)
+
+        # auth and copy env
+        job.image(image_path('aip'))
+        authenticate_cloud_credentials_in_job(job)
+        copy_common_env(job)
+        hpo_panels = str(inputs.as_dict(dataset, GeneratePanelData)['hpo_panels'])
+        hail_inputs = inputs.as_dict(dataset, RunHailFiltering)
+        panel_input = str(inputs.as_dict(dataset, QueryPanelapp)['panel_data'])
+        # peddy can't read cloud paths
+        local_ped = get_batch().read_input(str(hail_inputs['pedigree']))
+        labelled_vcf = str(hail_inputs['labelled_vcf'])
+        out_json_path = str(self.expected_outputs(dataset)['summary_json'])
+        input_mt = get_config()['workflow'].get(
+            'matrix_table', query_for_latest_mt(dataset.name)
+        )
+        job.command(
+            f'python3 reanalysis/validate_categories.py '
+            f'--labelled_vcf {labelled_vcf} '
+            f'--out_json {out_json_path} '
+            f'--panelapp {panel_input} '
+            f'--pedigree {local_ped} '
+            f'--input_path {input_mt} '
+            f'--participant_panels {hpo_panels} '
+        )
+        expected_out = self.expected_outputs(dataset)
+        return self.make_outputs(dataset, data=expected_out, jobs=job)
+
+
+def _aip_html_meta(
+    output_path: str,  # pylint: disable=W0613:unused-argument
+) -> dict[str, str]:
+    """
+    Add meta.type to custom analysis object
+    This isn't quite conformant with what AIP alone produces
+    e.g. doesn't have the full URL to the results in GCP
+    """
+    return {
+        'type': 'aip_output_json',
+        'is_singleton': '',  # not doing this at the moment
+        'is_exome': get_config()['workflow'].get('sequencing_type') == 'exome'
+    }
+
+
+@stage(
+    required_stages=[ValidateMOI],
+    analysis_type='custom',
```

Comment:
Ditto

Reply:
Currently Custom is consistent with previous AIP runs, I've like to get on making a new specific type for these reports

Reply:
Should we do this as part of this change? It's easy to add new analysis types, and we can migrate old entries very quickly (one SQL update).

Reply:
I'm open to it... could we create two entry types? One for the report JSON, one for the HTML?

Reply:
Yeah for sure, it's just a value in a table (with a foreign key), there's less friction, so avoiding overloading existing types I think is a better approach here